### DISCUSSION TITLE
Is there a way to get the namespace of the target method?

### DISCUSSION THREAD

**Discussion by craigbehnke** [metalama/Metalama#35] [Q&A] [answered]
Hello,

I'm currently working on migrating tracing functionality over to Metalama, but I'm running into issues getting a fully-defined method name for the target method.

As an example, here's some code:
```
namespace MyApp.Database  
{ 
    public class MyDatabaseInterface  
    {  
        [MyAspect]
        public string GetUsername(string id)
        {
            // my code here
        }
    }
}
```

Meanwhile, my Aspect code looks something like this (pulling from the docs):

```
namespace Doc.SimpleLogging
{
    public class SimpleLogAttribute : OverrideMethodAspect
    {
        public override dynamic? OverrideMethod()
        {
            Console.WriteLine( $"Entering {meta.Target.Method}" );

            try
            {
                return meta.Proceed();
            }
            finally
            {
                Console.WriteLine( $"Leaving {meta.Target.Method}" );
            }
        }
    }
}
```

What I expect/want to see is `Entering MyApp.Database.MyDatabaseInterface.GetUsername`, but instead I see `Entering MyDatabaseInterface.GetUsername`.

Is there a way to get the long form of the target at this time?

My guess at what the solution may look like is to prepend the namespace to the method (hence my title) with code like `$"Entering {getNamespace()}.{meta.Target.Method}"`, but I haven't found a way to do this yet.

Thanks in advance for your help!

**Comment by prochan2** [comment, marked as answer]
You can compose the fully qualified name as `$"{meta.Target.Method.DeclaringType.FullName}.{meta.Target.Method.Name}"`.

Alternatively, you can use `meta.Target.Method.ToDisplayString()`. See https://doc.metalama.net/api/metalama_framework_code_idisplayable for details about optional parameters of this method.

**Reply by craigbehnke** [reply]
That solved it. Thank you very much!
